### PR TITLE
Added view for logger to trigger from clicking Generate PDF button

### PR DESCRIPTION
**Pull request by Tanvez**
- Added script function in syllabuspdf index.html file to trigger from onclick event
- Added logger_view endpoint to receive the onclick event and log course_id and user_id
- Created test logger_view

Note: I tried to write the script in the base.html but it did not seem to trigger from the syllabuspdf index.html file

### CODE REVIEW

File: syllabuspdf/views.py [hunk 1]
```diff
@@ -100,3 +100,31 @@ def index(request):
...
+        logger.info(f"Syllabus index: Syllabus PDF Generated: course_id={course_id} user_id={user_id}")
+        return HttpResponse(status=200)
+    except Exception as e:
+        err1,err2 = e.args
+        if err1 and err2:
+            logger.error(f"Syllabus index: Syllabus PDF Not Generated: canvas {err1} and {err2} not found in session")
+            return HttpResponseNotFound(f'{err1} and {err2} not found')
+        if err1:
+            logger.error(f"Syllabus index: Syllabus PDF Not Generated: canvas {err1} not found in session")
+            return HttpResponseNotFound(f'{err1} not found')
```
Comment (dodget) [inline syllabuspdf/views.py]: The try/except then conditional block is a bit chunky. Looks like you are checking for both session attributes, then checking for each one individually? What do you think of something like:
```py
try:
    course_id = request.session['course_id']
    user_id = request.session['user_id']
    logger.info(f"Syllabus index: Syllabus PDF Generated: course_id={course_id} user_id={user_id}")
    return HttpResponse(status=200)
except KeyError as e: ...the rest of your code
```
That would condense it a bit, and still get at the fact that both `course_id` and `user_id` should be in `request.session`

**Review by dodget — COMMENTED** [review]
I have a question on the approach here. Does this tool have a back-end function/method that calls the canvas api or does the syllabus generation? If so, couldn't we just add some logging there? If not, this is a clever approach.

**Comment by Tanvez** [comment]
> I have a question on the approach here. Does this tool have a back-end function/method that calls the canvas api or does the syllabus generation? If so, couldn't we just add some logging there? If not, this is a clever approach.

Great question! Yea, the tool does in the `apirequest.py`. There are functions that call the canvas api that are used in the views. These views/canvas api request functions are called when the tool is initially navigated to and is logged. The syllabus generation from what I can see uses the native window.print() function which is not attached to an api call and isn't logged. Hope that answers your question!

File: syllabuspdf/views.py (hunk 1 above)
Comment (Tanvez) [reply syllabuspdf/views.py]: oh i see! that looks much shorter than what I wrote, Thanks Tylor. Will look into updating the code with this.

**Comment by Tanvez** [comment]
todo: update to post in logger endpoint

**Review by dodget — APPROVED** [review]
👍

**Review by arthurian — APPROVED** [review]
Thanks for the live walkthrough today @Tanvez. Looks good!

**Pull request merged**